src/swap/components/SwapAmountInputV2.tsx

```diff
@@ -0,0 +1,51 @@
+import { useCallback, useContext, useMemo } from 'react';
+
+import { isValidAmount } from '../utils';
+import { TokenChip } from '../../token';
+import { SwapContext } from '../context';
+import type { SwapAmountInputReact } from '../types';
+
+export function SwapAmountInputV2({ label, token, type }: SwapAmountInputReact) {
+  const { fromAmount, setFromAmount, toAmount, setToAmount } = useContext(SwapContext);
+
+  const amount: string = useMemo(() => {
+    if (type === 'to') {
+      return toAmount;
+    }
+    return fromAmount;
+  }, [type, toAmount, fromAmount]);
+
+  const setAmount: (a: string) => void = useMemo(() => {
+    if (type === 'to') {
+      return setToAmount;
+    }
+    return setFromAmount;
+  }, [type, setToAmount, setFromAmount]);
+
+  const handleAmountChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    if (isValidAmount(event.target.value)) {
+      setAmount?.(event.target.value);
+    }
+  }, []);
+
+  return (
+    <div
+      className="box-border flex w-full flex-col items-start gap-[11px] border-b border-solid  bg-[#FFF] p-4"
```

Comment:
Btw when you have very long `className` you can use the `cn` utility we have from `/lib/utils`, to organize your classes and do something like.

```
className={cn('box-border flex w-full flex-col items-start ',
 'gap-[11px] border-b border-solid  bg-[#FFF] p-4')}
```

so it's not so long